### PR TITLE
Convert plank cutting to player action - Ready

### LINKED ISSUE #32587: Game going non-responsive after Cut Tree Trunk into Planks action is performed.
Game version: 0.D-5823-g763c25c (tiles)

Operating system: Win 10 Pro Version 10.0.17763

Tiles or curses: tiles

Mods active:
Dark Days Ahead
Disable NPC Needs
Makeshift Items Mod
More Survival Tools
Modular Turrets
Salvaged Robots
Hydroponics
Mining Mod
Jury-Rigged Robots
StatsThoughSkills IV

### Expected behavior
Generate planks and other wood items from a tree trunk.
Game continues to run after action completes, allowing you to repeat the action or do something else in game.

### Actual behavior
After planks and other wood items generate, with in game time passing in 5 minute increments, and the message screen showing what was created, the in game time continues to progress as if the Period key "." was being held down. 
After 2-10 seconds game goes non-responsive and must be force closed.

### Steps to reproduce the behavior
Press Shift-8 or asterisk "*" to open Construction Menu.
Find Cut Tree Trunk into Planks in the menu with a with Stone Adze or Stone Hand Axe wielded or in your inventory.
Select direction of the tree trunk or tree stump to start the action.
Wait for action to complete, wood products generate in the message log, and the in game time to pass until game crashes.

### Additional notes
Tested multiple times on two computers across a few different builds, 9363, 9365, and 9366 that released on 7/23. The latest test being on Build 9366 with a fresh world and character, world settings and mods copied from a pre-existing world. 

I believe this was working for me on one of the builds from 7/21 or earlier.

### PR DESCRIPTION
**Pull request by macrosblackd**
#### Summary

```SUMMARY: Bugfixes "Convert plank cutting to player action"```

#### Purpose of change
Fixes #32587 - Move the logic and consumption of time of chopping a log into planks to an action finisher.

#### Describe the solution

Create a new action type, ```ACT_CHOP_PLANKS```.
Move the logic behind the creation of planks and scraps to a new action finisher.
Assign the new action when the calling from lumber actions (direct use of axe or saw) and when using the construction action (tree trunk -> planks, skips logs entirely).

### CODE REVIEW

**Comment by macrosblackd** [comment]
I'm currently experiencing a freeze when the game tries to spawn the planks in
```
if( planks > 0 ) {
        g->m.spawn_item( act->placement, "2x4", planks, 0, calendar::turn );
        p->add_msg_if_player( m_good, _( "You produce %d planks." ), planks );
    }
```

**Comment by anothersimulacrum** [comment]
I believe you need to add a `chop_planks_do_turn`, `chop_planks_finish` is getting called multiple times.

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -3566,6 +3568,28 @@ void activity_handlers::chop_logs_finish( player_activity *act, player *p )
...
+        g->m.spawn_item( act->placement, "2x4", planks, 0, calendar::turn );
+        p->add_msg_if_player( m_good, _( "You produce %d planks." ), planks );
+    }
+    if( scraps > 0 ) {
+        g->m.spawn_item( act->placement, "splinter", scraps, 0, calendar::turn );
+        p->add_msg_if_player( m_good, _( "You produce %d splinters." ), scraps );
+    }
+    if( planks < max_planks / 2 ) {
+        p->add_msg_if_player( m_bad, _( "You waste a lot of the wood." ) );
+    }
```
Comment (anothersimulacrum) [inline src/activity_handlers.cpp]:
```suggestion
    }
    act->set_to_null();
```

Comment (macrosblackd) [reply src/activity_handlers.cpp]: This fixed it!

**Comment by Night-Pryanik** [comment]
> I'm currently experiencing a freeze when the game tries to spawn the planks in

I guess that's because game is trying to spawn items in a non-container terrain like a tree trunk. You'd better use the original `i_add_or_drop` function for that.

**Comment by macrosblackd** [comment]
> > I'm currently experiencing a freeze when the game tries to spawn the planks in
> 
> I guess that's because game is trying to spawn items in a non-container terrain like a tree trunk. You'd better use the original `i_add_or_drop` function for that.

I got it fixed - apparently the action needed to be set to null after spawning the items.

**Pull request merged**